Shared/LayerCacheManager.cpp

```diff
@@ -291,6 +292,11 @@ void LayerCacheManager::layerToJson(Layer* layer)
   if (markupLayer)
     layerPath = markupLayer->path();
 
+  // Get KmlLayers
+  auto kmlLayer = dynamic_cast<KmlLayer*>(layer);
+  if (kmlLayer)
+    layerPath = kmlLayer->dataset()->url().toLocalFile();
```

Comment:
For now this will always be a local file, but is it worth checking to make sure the url is a local file first? That check would help in the future if we support online datasets, but may be overkill for now.